cyclonedx/model/definition.py

```diff
@@ -246,7 +243,7 @@ def parent(self) -> Optional[BomRef]:
 
     @parent.setter
     def parent(self, parent: Optional[Union[str, BomRef]]) -> None:
-        self._parent = _bom_ref_from_str(parent)
+        self._parent = _bom_ref_from_str(parent, optional=True)
```

Comment:
@hakandilek this was the needed fix for the flaky test.

a `BomRef` object without a `value` was created - which had an inconsistent hash value.
Correct would be to have no `BomRef` object at all, if no value was given.